### ISSUE TITLE
[Bug] Intermission screen times are out by one minute in Arcane Dimensions/Violent Rumble when cl_nocsqc is set to 0

### ISSUE THREAD

**Issue by korendian** [Novum/vkQuake#436]
Something i've noticed while recording various demos is that the level time on the intermission screen for maps in Arcane Dimensions and also the Violent Rumble mod are sometimes incorrect by a full minute. Any time that ends with 31-59 seconds has an extra minute added to the clock. For example a level time of 0:31 becomes 1:31 and a time of 1:59 becomes 2:59.

I've done some testing and discovered this does not occur when cl_nocsqc is set to 1. This can easily be replicated by exiting ad_start, ad_chapters in AD, or start in violent rumble after 31-59 seconds with this cvar set to 1 and then to 0. This same bug also occurs in the latest quakespasm-spiked and can be similarly fixed by setting pr_checkextension 0 to disable all CSQC. It does not occur in quakespasm.

I've only seen this occur with Arcane Dimensions & Violent Rumble. It does not occur with id1, copper, quoth or alkaline (which uses CSQC for an extended hud). I'm not sure if this is a bug with the csqc code used in these mods or the implementation. 

Tested with: [vkQuake 1.12.2_win64, clean install.](https://github.com/Novum/vkQuake/releases/download/1.12.2/vkquake-1.12.2_win64.zip)
Mods used:  [Arcane Dimensions with Patch 1](https://www.moddb.com/mods/arcane-dimensions/downloads/arcane-dimensions-180-with-patch-1) and [Violent Rumble](https://www.moddb.com/mods/violent-rumble/downloads/violent-rumble-100)
Platform: Windows 10 20H2

**Comment by jgreenburg** [comment]
I have also experienced issues with inaccurate level completion time on intermission screens.  I first noticed it under QSS on several Arcane Dimensions 1.81 playthroughs as well as vkQuake on my xmasJam 2021 playthroughs as well.  For me, times are inaccurate to varying degree, sometimes its over or even under by a few seconds, and sometimes its over more than a minute.  I do not have any [cl_no*] cvars listed in either my config or quake.rc(For AD/XmasJam), so I'm using the engine defaults.  Here is my [config](https://gist.github.com/jgreenburg/fddaf20a55c0d2c06020f24c85e46618) so you can see if there are any other cvars that may also be attributing to the issue.  I haven't independently verified that cl_nocsqc 1 resolves the issue for me yet.

**Comment by fabiolimamp** [comment]
Most likely a mod issue. When using CSQC, the ending screen in AD is generated/controlled by the mod's own code, including the elapsed time

**Comment by temx** [comment]
Yes, this is a mod issue, see `csqc_hud*.qc` in the mods' source code, function `Hud_Intermission`:
`Hud_DrawCharFont24(pos+[144, 64], sprintf("%3.0f:%02.0f", intermission_time/60, intermission_time%60));`
The floats are getting rounded (this is expected behavior).
